util.c

```diff
@@ -2379,11 +2391,13 @@ void reopen_mddev(int mdfd)
 	 */
 	char *devnm;
 	int fd;
+
```

Comment:
I think it is right way to make it. We have to keep mdfd in case if we cannot reopen the device, so it is better to close in only if we successed to reopen mddev.

```c
/**
 * reopen_mddev() - Reopen without O_EXCL, but keep the same fd numer.
 *@mdfd:  file descriptor with O_EXCL
 *
 * If it cannot reopen the device then mdfd is not reopened.
 */

void reopen_mddev(int mdfd)
{
	char *devnm = fd2devnm(mdfd);
	int fd = open_dev(devnm);

	if (!is_fd_valid(fd))
		return;

	dup2(fd, mdfd);

	close_fd(&fd);
}
```